### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by Estherstarr**
In this PR, I implemented the following:-
-  [ ]  Set up associations between models
-  [ ] Users
 - A method that returns the 3 most recent posts for a given user.
- [ ] Posts
 - A method that updates the posts counter for a user.
 - A method which returns the 5 most recent comments for a given post.
- [ ] Comments
 - A method that updates the comments counter for a post.
- [ ] Likes
 - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :comments
+  has_many :posts
+  has_many :likes
```
Comment (Whoistolu) [inline app/models/user.rb]: - [ ] [REQUIRED] Great job with the associations @Estherstarr 👍🏽 . However, you need to add foreign _keys to the user_model.  If you take a look at the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) you will see that the user_model has `'author_id'`  as a foreign_key that shows its relationship with comment, like and post. Kindly implement that before asking for another review.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
...
+  has_many :likes
+
+  def recent_comments
+    comments.order(created_at: :DESC).limit(5)
+  end
+
+  def update_post_counter
+    user.update(posts_counter: user.posts.count)
+  end
 end
```
Comment (Whoistolu) [inline app/models/post.rb]: - [ ] [OPTIONAL] You can also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private.

```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

### This method can also be applied to the comment and the like model as well.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Estherstarr 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @Estherstarr 👋, 

Well done for implementing all the required changes and even the optional one as well 👏 
Your project is now complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**